MF/V3/Settings/Software.proto

```diff
@@ -19,4 +19,10 @@ message Software
     
     // Request available software packages.  If undefined all available packages are returned.
     repeated Package available = 2;
+
+    // Request upgrade to latest major version.
+    optional bool majorUpgrade = 3;
+
+    // Permit nightly release upgrades.
+    optional bool nightlyIncluded = 4;
```

Comment:
I would rename `nightlyIncluded` to `nightlyUpgrade` to match `'majorUpgrade`.  But not a big deal.